Farm&Gun/Assets/Scripts/Entities/Opponents/OpponentChaseState.cs

```diff
@@ -10,30 +11,45 @@ public class OpponentChaseState : State
     private OpponentHitState opponentHitState;
 
     [SerializeField]
-    private GameObject player;
+    private NavMeshAgent agent;
 
     [SerializeField]
-    private NavMeshAgent agent;
+    private float coroutineTime = 2f;
+
+    public GameObject Target { set; get; }
+
+    private Opponent opponent;
 
     private void Awake()
     {
-        player = GameObject.FindGameObjectWithTag("Player");
+        opponent = gameObject.transform.parent.parent.GetComponent<Opponent>();
+        Target = opponent.FindTheNearestAnimalToChase();
     }
     public override State RunCurrentState()
     {
-        if (gameObject.transform.parent.parent.GetComponent<Opponent>().IsEating &&
-            !gameObject.transform.parent.parent.GetComponent<Opponent>().IsHit)
+        if (opponent.IsEating && !opponent.IsHit)
         {
             return opponentEatState;
         }
-        else if (gameObject.transform.parent.parent.GetComponent<Opponent>().IsHit)
+        else if (opponent.IsHit)
```

Comment:
When there is return in a previous if statement then "else if" is not needed -> it can be  just a normal if

Reply:
oki :D